.github/workflows/ci.yml

```diff
@@ -102,14 +102,13 @@ jobs:
       - run: ${{ matrix.make_command }}
         shell: bash
 
-  # Builds for binary releases.
+  # Builds for binary releases. Runs on each build, but only uploads the artifacts on a tag starting with 'v'
```

Comment:
Why do we need to build this on every build?

Reply:
I think it makes sense to do so. Otherwise we'll only find out if it's broken when trying to push a tag :)  Plus it adds some extra coverage (e.g. win32 and linux32).

Reply:
We should detect any breakages (that are reasonable to detect) in the other job, the regular test job.

Reply:
Maybe we could remove some of the other builds to avoid duplication. E.g. only the nightly builds include glfw in the compilation, so the stable builds do nothing that the release-builds does not already do.

Reply:
Any specific reason why the release-builds cannot also function as a test-build? :)

Reply:
Maybe I'm seeing it wrong. I see us have 2 jobs: one for doing test builds and checking the correctness of code on CI, and another for releasing. You are saying that the former one doesn't check *all* the correctness we need.
So it's clear to me that the test job needs to be fixed/improved based on this, but instead you are suggesting to basically merge the two jobs together. This isn't necessary, unless there are blockers (which I'm not considering) for fixing the test job to do what you want.

Reply:
The main purpose of the release-builds is to create the binaries. In that process they (obviously) build the project, which in itself is a test - in fact the builds in the test-jobs (currently) don't do much more than performing a build :)

So one argument for running these on every build is that them working can be seen as a minimal signal that all is well, and the test-jobs can be used for more advanced tests. A second argument is that if we only run the release-builds when pushing a tag, it's easy for them to become outdated (and you only find out when pushing a tag).

Anyway, if you prefer to keep these more separate, I'll put back the `if`. I could prepare another PR to add test builds for win32 and linux32.

Reply:
> ll is well, and the test-jobs can be used for more advanced tests

One problem here is that the tests only care about the debug builds, reasonably assuming that release will have to build successfully if debug is fine. So if you combine it with the release job, we'll end up building both release and debug, wasting CI time. This isn't a problem right now, given the low rate of PRs, but it's not scaling well.

> A second argument is that if we only run the release-builds when pushing a tag, it's easy for them to become outdated (and you only find out when pushing a tag).

They got outdated now because we have a major change in the project (replacing cbindgen by bindgen). It's not supposed to happen often, and even when such breakage occurs, we can fix it as we see it: a breakage in the release job doesn't affect anybody.

I.e. the point of a CI check is to make sure that no bad code accidentally gets checked in. It's important be cause later the bad code will accumulate, and it may be hard to see when exactly things broke. If all we have broken is a release job, then the breakage is limited to this one tiny build script, which can be fixed or even rewritten.

If I understand correctly, what you want is coverage for 32-bit platform builds. This is reasonable, even if niche. How do you feel about the following proposal: we add a 3rd job to build these targetr, but we'll only run it on `master` merges (not any PRs). So we'll know quickly enough if something is broken, but we aren't going to stress the CI to build them on each PR iteration.

Reply:
Fair points. Ok, I added the `if` back so they only run on tag pushes. I'll have a look at the test-builds in another PR. I like the idea of running certain builds only on pushed to master.

Reply:
Thank you! Let's adjust the comment as well. It currently says that the release builds happen all the time.

Reply:
Done!